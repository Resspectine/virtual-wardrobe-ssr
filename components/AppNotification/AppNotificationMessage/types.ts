import { Notification } from '@/types/notification';

export interface AppNotificationMessageProps {
  notification: Notification;
  onRemove: (id: number) => void;
}
