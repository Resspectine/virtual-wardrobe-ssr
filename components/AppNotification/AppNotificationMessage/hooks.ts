import { useCallback, useEffect, useRef, useState } from 'react';

import { AppNotificationMessageProps } from './types';

export const useAppNotificationMessage = ({ notification, onRemove }: AppNotificationMessageProps) => {
  const timerRef = useRef<ReturnType<typeof setTimeout> | null>(null);
  const animationRef = useRef<ReturnType<typeof setTimeout> | null>(null);
  const [isRemoving, setRemoving] = useState(false);

  const clearRefs = useCallback(() => {
    if (timerRef.current) {
      clearTimeout(timerRef.current);
    }

    if (animationRef.current) {
      clearTimeout(animationRef.current);
    }
  }, []);

  useEffect(() => {
    timerRef.current = setTimeout(() => onRemove(notification.id), 3000);
    animationRef.current = setTimeout(() => setRemoving(true), 2600);

    return clearRefs;
  }, []);

  const onCancelClick = useCallback(
    (event: React.MouseEvent<SVGSVGElement, MouseEvent>) => {
      event.stopPropagation();

      setRemoving(true);

      timerRef.current = setTimeout(() => onRemove(notification.id), 400);
    },
    [setRemoving, onRemove]
  );

  return {
    onClick: clearRefs,
    isRemoving,
    onCancelClick,
  };
};
