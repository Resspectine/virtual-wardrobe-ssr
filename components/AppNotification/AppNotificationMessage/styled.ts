import { Cancel } from '@mui/icons-material';
import DoneIcon from '@mui/icons-material/Done';
import MaterialErrorIcon from '@mui/icons-material/Error';
import ReportIcon from '@mui/icons-material/Report';
import SyncIcon from '@mui/icons-material/Sync';
import { Box, styled } from '@mui/system';

import { NotificationType } from '@/types/notification';

const variants: Record<
  NotificationType,
  {
    color: string;
    backgroundColor: string;
    borderColor: string;
  }
> = {
  notification: {
    backgroundColor: '#fff',
    borderColor: '#7fbffe',
    color: '#000',
  },
  warning: {
    backgroundColor: '#fff',
    borderColor: '#ffeb3b',
    color: '#000',
  },
  error: {
    backgroundColor: '#fff',
    borderColor: '#f44336',
    color: '#000',
  },
  loading: {
    backgroundColor: '#fff',
    borderColor: '#8561c5',
    color: '#000',
  },
  success: {
    backgroundColor: '#fff',
    borderColor: '#a2cf6e',
    color: '#000',
  },
};

export const AppNotificationMessageWrapper = styled(Box)<{ remove: boolean; type: NotificationType }>(
  ({ remove, type }) => ({
    display: 'flex',
    alignItems: 'flex-start',
    cursor: 'pointer',
    padding: '5px 10px',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 6,
    animation: !remove ? 'enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29)' : 'left 500ms linear',
    marginBottom: 10,
    '@keyframes enter': {
      '0%': {
        transform: 'translateX(-100%)',
      },
      '100%': {
        transform: 'translateX(0)',
      },
    },
    '@keyframes left': {
      '0%': { transform: 'translateX(0)', opacity: 1 },
      '100%': { transform: 'translateX(100%)', opacity: 0 },
    },
    ...variants[type],
  })
);

export const AppNotificationMessageCancel = styled(Cancel)({
  cursor: 'pointer',
  marginLeft: 5,
  color: '#000',
});

export const LoadingIcon = styled(SyncIcon)({
  animation: 'rotate 1s linear infinite',
  color: '#8561c5',
  marginRight: 5,
  '@keyframes rotate': {
    '0%': { transform: 'rotate(360deg)' },
    '100%': { transform: 'rotate(0deg)' },
  },
});

export const SuccessIcon = styled(DoneIcon)({
  marginRight: 5,
  color: '#a2cf6e',
});

export const ErrorIcon = styled(ReportIcon)({
  marginRight: 5,
  color: '#f44336',
});

export const NotificationIcon = styled(MaterialErrorIcon)({
  marginRight: 5,
  color: '#7fbffe',
});

export const WarningIcon = styled(MaterialErrorIcon)({
  marginRight: 5,
  color: '#ffeb3b',
});
