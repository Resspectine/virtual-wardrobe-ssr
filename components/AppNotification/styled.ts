import { Box, styled } from '@mui/system';

export const AppNotificationRoot = styled(Box)({
  position: 'fixed',
  top: 40,
  left: 20,
  zIndex: 3000,
});
