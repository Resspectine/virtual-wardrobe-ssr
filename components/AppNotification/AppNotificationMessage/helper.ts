import { StyledComponent } from '@emotion/styled';

import { ErrorIcon, LoadingIcon, NotificationIcon, SuccessIcon, WarningIcon } from './styled';

import { NotificationType } from '@/types/notification';

export const iconByType: Record<NotificationType, StyledComponent<Record<string, unknown>>> = {
  notification: NotificationIcon,
  error: ErrorIcon,
  loading: LoadingIcon,
  success: SuccessIcon,
  warning: WarningIcon,
};
