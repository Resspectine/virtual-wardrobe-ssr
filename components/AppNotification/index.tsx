import { FC } from 'react';

import { AppNotificationMessage } from './AppNotificationMessage';
import { AppNotificationRoot } from './styled';

import { useAppNotification } from '@/store/appNotification';

export const AppNotification: FC = () => {
  const notifications = useAppNotification(state => state.notifications);
  const deleteNotification = useAppNotification(state => state.deleteNotification);

  return (
    <AppNotificationRoot>
      {notifications.map(notification => (
        <AppNotificationMessage onRemove={deleteNotification} notification={notification} key={notification.id} />
      ))}
    </AppNotificationRoot>
  );
};
