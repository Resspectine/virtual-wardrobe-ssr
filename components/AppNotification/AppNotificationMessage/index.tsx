import { FC } from 'react';

import { iconByType } from './helper';
import { useAppNotificationMessage } from './hooks';
import { AppNotificationMessageCancel, AppNotificationMessageWrapper } from './styled';
import { AppNotificationMessageProps } from './types';

export const AppNotificationMessage: FC<AppNotificationMessageProps> = ({ onRemove, notification }) => {
  const { onClick, isRemoving, onCancelClick } = useAppNotificationMessage({
    notification,
    onRemove,
  });

  const Icon = iconByType[notification.type];

  return (
    <AppNotificationMessageWrapper type={notification.type} remove={isRemoving} onClick={onClick}>
      <Icon />
      {notification.message}
      <AppNotificationMessageCancel data-testid="cancel-app-notification" onClick={onCancelClick} />
    </AppNotificationMessageWrapper>
  );
};
