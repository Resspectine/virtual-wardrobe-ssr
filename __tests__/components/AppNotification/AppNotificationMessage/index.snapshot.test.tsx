/* eslint-disable no-plusplus */
import { render } from '@testing-library/react';

import { notificationBuilder } from './index.test';

import { AppNotificationMessage } from '@/components/AppNotification/AppNotificationMessage';

beforeEach(() => {
  jest.useFakeTimers();
});

afterEach(() => {
  jest.useRealTimers();
});

describe('FormSubmitSection', () => {
  test('should be success and match snapshot', () => {
    const notification = notificationBuilder({
      overrides: {
        type: 'error',
        message: 'error',
      },
    });

    const { asFragment } = render(<AppNotificationMessage notification={notification} onRemove={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should be success and match snapshot', () => {
    const notification = notificationBuilder({
      overrides: {
        type: 'loading',
        message: 'loading',
      },
    });
    const { asFragment } = render(<AppNotificationMessage notification={notification} onRemove={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should be success and match snapshot', () => {
    const notification = notificationBuilder({
      overrides: {
        type: 'notification',
        message: 'notification',
      },
    });
    const { asFragment } = render(<AppNotificationMessage notification={notification} onRemove={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should be success and match snapshot', () => {
    const notification = notificationBuilder({
      overrides: {
        type: 'success',
        message: 'success',
      },
    });
    const { asFragment } = render(<AppNotificationMessage notification={notification} onRemove={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should be success and match snapshot', () => {
    const notification = notificationBuilder({
      overrides: {
        type: 'warning',
        message: 'warning',
      },
    });
    const { asFragment } = render(<AppNotificationMessage notification={notification} onRemove={() => {}} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
