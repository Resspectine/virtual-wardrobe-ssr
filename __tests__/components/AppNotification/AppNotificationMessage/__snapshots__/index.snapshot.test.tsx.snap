// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmitSection should be success and match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 10px;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  max-width: 350px;
  -webkit-animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  margin-bottom: 10px;
  background-color: #fff;
  border-color: #f44336;
  color: #000;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
  color: #f44336;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  cursor: pointer;
  margin-left: 5px;
  color: #000;
}

<div
    class="MuiBox-root emotion-0"
    type="error"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="ReportIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.73 3H8.27L3 8.27v7.46L8.27 21h7.46L21 15.73V8.27L15.73 3zM12 17.3c-.72 0-1.3-.58-1.3-1.3 0-.72.58-1.3 1.3-1.3.72 0 1.3.58 1.3 1.3 0 .72-.58 1.3-1.3 1.3zm1-4.3h-2V7h2v6z"
      />
    </svg>
    error
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="cancel-app-notification"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`FormSubmitSection should be success and match snapshot 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 10px;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  max-width: 350px;
  -webkit-animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  margin-bottom: 10px;
  background-color: #fff;
  border-color: #8561c5;
  color: #000;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  -webkit-animation: rotate 1s linear infinite;
  animation: rotate 1s linear infinite;
  color: #8561c5;
  margin-right: 5px;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  cursor: pointer;
  margin-left: 5px;
  color: #000;
}

<div
    class="MuiBox-root emotion-0"
    type="loading"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="SyncIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4V1L8 5l4 4V6c3.31 0 6 2.69 6 6 0 1.01-.25 1.97-.7 2.8l1.46 1.46C19.54 15.03 20 13.57 20 12c0-4.42-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6 0-1.01.25-1.97.7-2.8L5.24 7.74C4.46 8.97 4 10.43 4 12c0 4.42 3.58 8 8 8v3l4-4-4-4v3z"
      />
    </svg>
    loading
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="cancel-app-notification"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`FormSubmitSection should be success and match snapshot 3`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 10px;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  max-width: 350px;
  -webkit-animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  margin-bottom: 10px;
  background-color: #fff;
  border-color: #7fbffe;
  color: #000;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
  color: #7fbffe;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  cursor: pointer;
  margin-left: 5px;
  color: #000;
}

<div
    class="MuiBox-root emotion-0"
    type="notification"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="ErrorIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </svg>
    notification
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="cancel-app-notification"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`FormSubmitSection should be success and match snapshot 4`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 10px;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  max-width: 350px;
  -webkit-animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  margin-bottom: 10px;
  background-color: #fff;
  border-color: #a2cf6e;
  color: #000;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
  color: #a2cf6e;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  cursor: pointer;
  margin-left: 5px;
  color: #000;
}

<div
    class="MuiBox-root emotion-0"
    type="success"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="DoneIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
    success
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="cancel-app-notification"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`FormSubmitSection should be success and match snapshot 5`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 5px 10px;
  border-width: 1px;
  border-style: solid;
  border-radius: 6px;
  max-width: 350px;
  -webkit-animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  animation: enter 300ms cubic-bezier(0.15, 0.7, 0.71, 1.29);
  margin-bottom: 10px;
  background-color: #fff;
  border-color: #ffeb3b;
  color: #000;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  margin-right: 5px;
  color: #ffeb3b;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  cursor: pointer;
  margin-left: 5px;
  color: #000;
}

<div
    class="MuiBox-root emotion-0"
    type="warning"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="ErrorIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </svg>
    warning
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
      data-testid="cancel-app-notification"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
